tests/util/formatter.py

```diff
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 SciCat Project (https://github.com/SciCatProject/scitacean)
+
+from scitacean import Dataset
+from scitacean.util.formatter import DatasetPathFormatter
+
+
+def test_dataset_formatter_uses_dataset_fields():
+    dset = Dataset(type="raw", owner="magrat")
+    fmt = "{type} {owner}"
+    formatted = DatasetPathFormatter().format(fmt, dset)
+    expected = fmt.format(type=dset.type, owner=dset.owner)
+    assert formatted == expected
+
+
+def test_dataset_formatter_can_access_attrs_of_fields():
+    dset = Dataset(type="raw", _pid="prefix/actual-id")
+    formatted = DatasetPathFormatter().format("{pid.pid}", dset)
```

Comment:
why is there 2 times `pid` here?

Reply:
Because the `PID` class has two members, `prefix` and `pid`. I'm not happy with the latter but couldn't come up with something better.
So this code uses the `pid` field, not the `prefix`.